src/iproject.ts

```diff
@@ -93,27 +95,34 @@ export class IProject {
     this.state = state;
   }
 
-  public async getBuildMap() {
-    let buildMap: Map<string, IBMiJsonT> = new Map();
+  public async getBuildMap(): Promise<Map<string, IBMiJsonT>> {
+    if (!this.buildMap) {
+      return await this.updateBuildMap();
+    }
+    return this.buildMap;
+  }
+
+  public async updateBuildMap() {
+    this.buildMap.clear();
 
     const ibmiJsonPaths = await workspace.findFiles('**/.ibmi.json');
     for await (const ibmiJsonPath of ibmiJsonPaths) {
       try {
         const ibmiJsonContent: IBMiJsonT = JSON.parse((await workspace.fs.readFile(ibmiJsonPath)).toString());
         if (ibmiJsonContent && ibmiJsonContent.build) {
-          buildMap.set(path.dirname(ibmiJsonPath.fsPath), ibmiJsonContent);
+          this.buildMap.set(path.dirname(ibmiJsonPath.fsPath), ibmiJsonContent);
         }
       } catch { }
     };
 
-    if (!buildMap.has(this.workspaceFolder.uri.fsPath)) {
+    if (!this.buildMap.has(this.workspaceFolder.uri.fsPath)) {
       const unresolvedState = await this.getUnresolvedState();
       if (unresolvedState && unresolvedState.objlib) {
-        buildMap.set(this.workspaceFolder.uri.fsPath, { build: { objlib: unresolvedState.objlib } });
+        this.buildMap.set(this.workspaceFolder.uri.fsPath, { build: { objlib: unresolvedState.objlib } });
```

Comment:
I assume the logic to inherit attributes from the parent directory is somewhere else?

Reply:
Using this `buildMap` other extensions can implement the logic to inherit attributes from the parent directory. Currently in the Project Explorer the `buildMap` is only used for retrieving variables so this logic is not needed here.

We have that logic in the IBM i Developer.

Reply:
I see, I thought it would be good if the Project Explorer would implement these defined semantics of the project so that the target library and ccsid could be queried.   Since no one is depending on this yet, we can leave it as a TODO

Reply:
Good point. I have updated this PR to include a method to recursively search the parent directories of the buildMap to get the `.ibmi.json` information given a `Uri`.
```typescript
public async getIbmiJson(ibmiJsonUri: Uri, buildMap?: Map<string, IBMiJsonT>): Promise<IBMiJsonT | undefined>
```